MiniProject1-master/Holmes_Mini_Project_1.py

```diff
@@ -0,0 +1,89 @@
+from pattern.en import sentiment
+from pattern.en import modality
+
+def sort_data(data):
+    fin = open(data)
+    d = {}
+    names = []
+    speech = []
+    for line in fin:
+        line1 = line.replace(",,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,","",1)
+        line2 = line1.replace("\x92", "'")
+        line3 = line2.replace("\x85", "")
+        text = line3.replace("\r\n", "")
+        key = text[0:text.index(',')]
+        if key.strip() in d:
```

Comment:
You call key.strip() multiple times in the same loop; it's more efficient to, say, have key be the .strip()'d version, and then use it directly. Also, just a heads-up: entries will vary based on case and capitalization. If this isn't intended, I'd recommend converting keys all to the same case (e.g., lowercase)--python has a built-in method for that.